model/src/main/scala/org/zalando/nakadi/client/model/Model.scala

```diff
@@ -0,0 +1,192 @@
+ * @param title
+ */
+case class Event(
+  eventType: EventType,
+  additionalProperties: Boolean,
+  title: String)
+
+  
+  
+/**
+ * Metadata for this Event. Contains commons fields for both Business and DataChange Events. Most are enriched by Nakadi upon reception, but they in general MIGHT be set by the client.
+ * @param eid Identifier of this Event. Clients are allowed to generate this and this SHOULD be guaranteed to be unique from the perspective of the producer. Consumers MIGHT use this value to assert uniqueness of reception of the Event.
+ * @param eventType The EventType of this Event. This is enriched by Nakadi on reception of the Event based on the endpoint where the Producer sent the Event to. If provided MUST match the endpoint. Failure to do so will cause rejection of the Event.
+ * @param occurredAt Timestamp of creation of the Event generated by the producer.
+ * @param receivedAt Timestamp of the reception of the Event by Nakadi. This is enriched upon reception of the Event. If set by the producer Event will be rejected.
+ * @param parentEids Event identifier of the Event that caused the generation of this Event. Set by the producer.
+ * @param flowId The flow-id of the producer of this Event. As this is usually a HTTP header, this is enriched from the header into the metadata by Nakadi to avoid clients having to explicitly copy this.
+ * @param metadata This Metadata contains common fields unrelated to Nakadi logic. Should be mainly enriched by the Consumer.
+ *
+ */
+case class EventMetadata(
+  eid: String,
+  eventType: String,
+  occurredAt: String,
+  receivedAt: String,
+  parentEids: Seq[String],
+  flowId: String,
+  metadata: Map[String, Any])
+
+/**
+ * A Business Event. Usually represents a status transition in a Business process.
+ *
+ * @param metadata Metadata for this Event. Contains commons fields for both Business and DataChange Events. Most are enriched by Nakadi upon reception, but they in general MIGHT be set by the client.
+ */
+case class BusinessEvent(metadata: EventMetadata)
+
+/**
+ * Indicators of a `DataChangeEvent`'s referred data type and the type of operations done on them.
+ * @param dataType The datatype of the `DataChangeEvent`.
+ * @param dataOp The type of operation executed on the entity. * C: Creation * U: Update * D: Deletion * S: Snapshot
+ */
+case class DataChangeEventQualifier(
+  dataType: String,
+  dataOperation: DataOperation.Value)
+
+/**
+ * A Data change Event. Represents a change on a resource.
+ *
+ * @param data The payload of the type
+ * @param eventQualifier Indicators of a `DataChangeEvent`'s referred data type and the type of operations done on them.
+ * @param metadata Metadata for this Event. Contains commons fields for both Business and DataChange Events. Most are enriched by Nakadi upon reception, but they in general MIGHT be set by the client
+ */
+case class DataChangeEvent[T](
+  data: T,
+  eventQualifier: DataChangeEventQualifier,
```

Comment:
payload has no attribute eventQualifier. you might project it as inheritance instead
https://github.com/zalando/nakadi/blob/nakadi-jvm/api/nakadi-event-bus-api.yaml#L940